pkg/kademlia/kademliaclient/kademliaclient.go

```diff
@@ -147,21 +147,27 @@ func (dialer *Dialer) FetchPeerIdentityUnverified(ctx context.Context, address s
 }
 
 // FetchInfo connects to a node and returns its node info.
-func (dialer *Dialer) FetchInfo(ctx context.Context, target pb.Node) (_ *pb.InfoResponse, err error) {
+func (dialer *Dialer) FetchInfo(ctx context.Context, target pb.Node) (_ *pb.InfoResponse, _ *identity.PeerIdentity, err error) {
 	defer mon.Task()(&ctx)(&err)
 	if !dialer.limit.Lock() {
-		return nil, context.Canceled
+		return nil, nil, context.Canceled
 	}
 	defer dialer.limit.Unlock()
 
 	conn, err := dialer.dialNode(ctx, target)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
+	}
+	p := &peer.Peer{}
+	_, err = conn.client.Ping(ctx, &pb.PingRequest{}, grpc.Peer(p))
```

Comment:
Since we are removing kademlia, there may not be a ping available.

Reply:
`transport.Client` `FetchPeerIdentity` can also fetch the peer identity.

Even, if it didn't exist, we would need to implement it.

Reply:
updated to use FetchPeerIdentity